core/rio/turtle/src/test/java/org/eclipse/rdf4j/rio/turtle/ArrangedWriterTest.java

```diff
@@ -70,6 +81,38 @@ public void testWriteSingleStatementWithSubjectNamespace() {
 		assertEquals(expectedResult, outputWriter.toString());
 	}
 
+	@Test
+	public void testWriteRepeatedInlineBlankNode() {
+		Model model = new ModelBuilder().subject(exNs + "subject")
+				.add(vf.createIRI(exNs, "rel1"), bnode)
+				.add(vf.createIRI(exNs, "rel2"), bnode)
+				.subject(bnode)
+				.add(RDFS.LABEL, "the bnode")
+				.build();
+
+		model.setNamespace(RDFS.NS);
+		model.setNamespace("ex", exNs);
+
+		StringWriter stringWriter = new StringWriter();
+		BufferedWriter writer = new BufferedWriter(stringWriter);
+		WriterConfig config = new WriterConfig();
+		config.set(BasicWriterSettings.INLINE_BLANK_NODES, true);
+		Rio.write(model, writer, RDFFormat.TURTLE, config);
+
+		String sep = System.lineSeparator();
+		String expectedResult = "@prefix ex: <http://example.org/> ." + sep +
+				"@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> ." + sep +
+				"@prefix xsd: <http://www.w3.org/2001/XMLSchema#> ." + sep + sep +
+				"ex:subject ex:rel1 [" + sep +
+				"      rdfs:label \"the bnode\"" + sep +
+				"    ];" + sep +
+				"  ex:rel2 [" + sep +
+				"      rdfs:label \"the bnode\"" + sep +
+				"    ] ." + sep;
+
+		assertEquals(expectedResult, stringWriter.toString());
```

Comment:
I like the structure of the test, but I'm not sure I agree that the expected output is correct - see also the comment I left on the issue.

Reply:
Hi
I have made the changes according to the discussion. There is no inlining for a repeated blank node. The test case has also been updated.